Test Project/Assets/Scripts/ToggleButtonColour.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ToggleButtonColour : MonoBehaviour
+{
+
+
+	#region Serialized Fields
+	
+	[SerializeField]
+	private List<Color> m_Colors;
+	[SerializeField] 
+	private Button m_TargetButton;
+
+	#endregion
+
+
+	#region Non-Serialized Fields
+
+	[System.NonSerialized]
+	private int m_NextColor = 1;
+
+	#endregion
+
+
+	#region Monobehavior Methods
+
+	private void Start()
+	{
+		SetColour(0);
+	}
+
+	#endregion
+
+
+	#region Methods
+
+	private void SetColour(int index)
+	{
+		if (this.m_TargetButton == null || this.m_Colors.Count <= index)
```

Comment:
We generally don't use `this` when accessing a field (the semantics are already clear). 

Our convention is to use `this` when accessing properties, because our property names start with an uppercase letter. In some cases, it's hard to tell in a block of code whether you're referencing a property or a class (i.e. `Foo.Bar` could mean `public object Foo { get; set; }` or it could mean `class Foo { public const string Bar; }`). Using `this` makes the semantics clear in that scenario.